File: Stack.cpp
#include <iostream>
#include <iterator>
#include <array>

using namespace std;

void push(int value);
void pop();
void display();

const int MAX = 5;
int stackArr[MAX];

class stacks{

	private:
		int top;
		//int stackArr[];

	public:
		stacks(){

			top = -1;
			//stackArr[] = {0 , 0 , 0 , 0 , 0}

		}

		void push(int value){

			if(top >= MAX - 1){

				cout << "Stack Overflow" << endl;

			}else{

				top++;
				stackArr[top] = value;

			}

		}

		void pop(){

			if(top == -1){

				cout << "Stack underflow" << endl;

			}else{

				cout << "----" << stackArr[top] << " was popped" << endl;
				top--;

			}

		}

		void display(){

				cout << endl;

			if(top == -1){

				cout << "No elements exists" << endl;

			}else{
				
				for(int i = top; i >= 0; i--){

					cout << stackArr[i] << endl;

				}
			
			}

		}

};

int main(){

	stacks arr;

	arr.push(4);
	arr.push(5);
	arr.push(1);
	arr.push(6);
	arr.push(7);
	arr.push(10);
	
	arr.pop();

	arr.display();

 	return 0;

}

File: Stacks/Stack.java
import java.util.Scanner;


class StackArr{

	static final int MAX = 10;
	static int[] arr;

	StackArr(){

		arr = new int[MAX];

		for(int i = 0; i < MAX; i++){

			arr[i] = -1;

		}

	}

	public static void addElement(int value){

		for(int i = 0; i < MAX; i++){

			if(arr[i] == -1){

				arr[i] = value;
				break;

			}

		}

	}

	public static void display(){

		System.out.println("\n");

		for(int i = MAX - 1; i > -1; i--){

			if(arr[i] != -1){

				System.out.println(arr[i]);

			}

		}

		System.out.println("\n");

	}

	public static void pop(){

		for(int i = MAX - 1; i > -1; i--){

			if(arr[i] != -1){

				arr[i] = -1;
				break;

			}

		}

	}

}

public class Stack{
	
	public static void main(String[] args){
	
		StackArr stack = new StackArr();

		stack.addElement(4);
		stack.addElement(7);
		stack.addElement(0);
		stack.addElement(11);
		stack.addElement(3);
		stack.addElement(8);
		stack.addElement(2);

		stack.display();

		stack.pop();
		stack.pop();
		stack.pop();

		stack.display();
	
	}

}